security/fma-message-passing-contracts.md

```diff
@@ -0,0 +1,140 @@
+    - `L1BlockInterop`: https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L2/L1BlockInterop.sol
+    - `SuperchainTokenBridge`: https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L2/SuperchainTokenBridge.sol
+
+
+> 🗂️
+> **For more context about the Interop project, refer to the following docs:**
+> 1. [Interop System Diagram](https://www.notion.so/16c8052fcbb24b93ad1a539b5f8db4c1?pvs=21)
+> 2. [Interop PID](https://www.notion.so/16c8052fcbb24b93ad1a539b5f8db4c1?pvs=21)
+> 3. [Interop Audit Request](https://docs.google.com/document/d/1Rcuzbsguh7koT2jFru5ft9T8zAvjBEzbt0zF5LNQQ08/edit?tab=t.0)
+
+
+## Failure Modes and Recovery Paths
+
+### FM1: Valid message is initiated but destination chain lacks origin chain in its dependency set
+
+- **Description:** A valid cross-chain message is initiated via `sendMessage` (or indirectly through contracts such as `SuperchainWETH` or `SuperchainTokenBridge`) with a specified destination chain ID. However, if the destination chain’s dependency set does not include the originating chain, the relay process (executed via `relayMessage`) will fail—preventing the message from being finalized.
+- **Risk Assessment:** High.
+    - Potential Impact: Critical. Users may lose access to assets such as `SuperchainWETH` or `SuperchainERC20` after burn/locked.
+    - Likelihood: High. `L2ToL2CrossDomainMessenger` doesn’t check whether the `destination` is a valid value that guarantees the message will be finalized there. Users can inadvertently target an arbitrary chain ID that is not recognized in the destination’s dependency set.
+- **Mitigations:** Currently, the op-interop set is intended to contain chains following a complete graph, meaning all chains include each other in their dependency set. This reduces the chance of equivocation given a known set of chain IDs. However, there are other approaches to consider:
+    - Ideally, there could be some pre-send validation. The `L2ToL2CrossDomainMessenger` (or `SuperchainTokenBridge`/`SuperchainWETH`) should be able to check against a list of chain IDs whose dependency set includes the chain where the message is about to be initiated. This was previously implemented by checking the `L1BlockInterop` dependency list, serving as a convenient method since the op-interop set follows a complete graph—ensuring that dependencies are identical across all chains. However, this code is being removed.
+    - Block builders/sequencers/`op-supervisor` may check if a transaction sent through the `L2ToL2CrossDomainMessenger` targets a `destination` that does not have the origin chain in its dependency set and drop it.
+    - There should be a way to claim back bridged funds, for example, only after the message expiration time has elapsed.
+    - Trusted bridge frontends should prevent users from sending tokens to unsupported chains.
```

Comment:
Let's clarify which of these we plan to implement (action items to track before production), and which are nice to haves that we might do after launch (tracked in issues somewhere?)